add scientific calcultor

ishanbagchi opened the pull request:
# add a Scientific Calculator

Add a command-line scientific calculator that does a little more than classical addition, subtraction, multiplication, and division. It calculates the sin, cos, tan, and much more like this.

Fixes: #125

## Add a scientific calculator in Python language and an instruction.md file

Please delete options that are not relevant.

-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

# Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generates no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [ ] Any dependent changes have been merged and published in downstream modules

kaustubhgupta commented on Python/Calculator/INSTRUCTIONS.md:
```diff
@@ -0,0 +1,38 @@
+
+you can use upper or lower case for operations    
+
+Operators|Functions
```
The readme needs proper alignment right now it's all messed up:
![image](https://user-images.githubusercontent.com/43691873/95007881-50e13500-0632-11eb-8142-a9e654e00619.png)

kaustubhgupta commented on Python/Calculator/calculator.py:
```diff
@@ -0,0 +1,85 @@
+'''
```
Remove this docstring as you have created a separated file for this

kaustubhgupta commented on Python/Calculator/calculator.py:
```diff
@@ -0,0 +1,85 @@
+import math
+from builtins import print
+
+print ("Please read instructions in the code\nor reach https://github.com/kaustubhgupta/Hacktoberfest2020_ for help \n*************************")
```
Remove this print statement

kaustubhgupta commented on Python/Calculator/calculator.py:
```diff
@@ -0,0 +1,85 @@
+from builtins import print
+
+print ("Please read instructions in the code\nor reach https://github.com/kaustubhgupta/Hacktoberfest2020_ for help \n*************************")
+firstNumber = float(input())
```
While taking input, print something

kaustubhgupta commented on Python/Calculator/calculator.py:
```diff
@@ -0,0 +1,85 @@
+secondNumber = float(input())
+
+# functions of the calculator
+if op == "+":
```
Implement a class-based code, move all this code to a class, and create separate functions for each operation. If the operator requires only one operand then don't take the second number input and directly output the result.

kaustubhgupta replied:
What about the class code?

kaustubhgupta reviewed (changes requested):
Please correct the following things

kaustubhgupta commented on Python/Calculator/calculator.py:
```diff
@@ -0,0 +1,85 @@
+elif op == "ln":
+    print ("ln(", secondNumber , ")= ", math.log(secondNumber))
+else:
+    print ("incorrect operator") 
```
Make the entry of this program in Python directory readme

ishanbagchi commented:
@kaustubhgupta sorry for the inconvenience. I'll correct it right away.

kaustubhgupta commented on Python/Calculator/calculator.py:
```diff
@@ -0,0 +1,63 @@
+    print(firstNumber, "%", secondNumber, "=", firstNumber % secondNumber )
+elif op == "!":
+    secondNumber = float(input("Enter the number: "))
+    theNumber = firstNumber = secondNumber 
```
What is this?

kaustubhgupta replied:
The line 39 doesn't seem right

ishanbagchi replied:
This is correct.

ishanbagchi commented:
@kaustubhgupta for this simple calculator program, if else is comparatively easy. So I thought of do it in this manner.

kaustubhgupta commented:
@ishanbagchi okay don't implement class code but fix the error I pointed.

ishanbagchi commented:
> @ishanbagchi okay don't implement class code but fix the error I pointed. 

Which error? I solved the error in INSTRUCTIONS.md, added the link in the language README.me, removed the print statement with the link. Anything else was there? 👀
